web/src/copy_and_paste.ts

```diff
@@ -640,6 +643,41 @@ export function maybe_transform_html(html: string, text: string): string {
     return html;
 }
 
+function add_text_and_select(text: string, $textarea: JQuery<HTMLTextAreaElement>): void {
+    const textarea = $textarea.get(0);
+    assert(textarea instanceof HTMLTextAreaElement);
+    const init_cursor_pos = textarea.selectionStart;
+    insertTextIntoField(textarea, text);
+    const new_cursor_pos = textarea.selectionStart;
+    textarea.setSelectionRange(init_cursor_pos, new_cursor_pos);
+}
+
+export function try_stream_topic_syntax_text(text: string): string | null {
+    const stream_topic = hash_util.decode_stream_topic_from_url(text);
+
+    if (!stream_topic) {
+        return null;
+    }
+    let avoid_transform = topic_link_util.will_produce_broken_stream_topic_link(
+        stream_topic.stream_name,
+    );
+    if (stream_topic.topic_name !== undefined) {
+        avoid_transform ||= topic_link_util.will_produce_broken_stream_topic_link(
+            stream_topic.topic_name,
+        );
+    }
+    if (avoid_transform) {
+        return null;
+    }
```

Comment:
I don't think you need this `avoid_transform` variable; just have `return null` in both code paths.

Reply:
Ironically, I had introduced that variable for readability.
Removed.